### DISCUSSION TITLE
Liquid templates custom objects

### DISCUSSION THREAD

**Discussion by di-effe** [leftwm/leftwm#629] [Ideas]
In the beginning I was very confused by liquid templates, but now I am very interested in them :D
Can we expand the existing objects?
I think that right now the leftwm filters are managed by leftwm-state.rs (fn template_handler), where I see window_title, workspace and localtime.

I am trying to understand if it is possible to add something on top of it without touching the leftwm code.
The idea would be for example to be able to add a filter that is the output of a checkupdates function.

**Comment by VuiMuich** [comment]
I am not sure if this is possible.
As far as I understand liquid templates this is just a filter to assist json (maybe actually language agnostic, idk) parsing.

**Reply by di-effe** [reply]
uhm too bad. 
At this point I am wondering if lemonbar for example can be managed both using the liquid template and appending other info.
It would be a pity to avoid to use the liquid stuff.

**Reply by VuiMuich** [reply]
Well if you call `leftwm state` from a script and grab its output into a variable, it should be easily possibe to append further info.

**Reply by di-effe** [reply]
I've to play around with it to understand it fully

**Comment by di-effe** [comment]
In the end I created a wrapper  script that loops combining the liquid template and custom scripts.
That's not very efficient but it works. Tonight I'll do it again using @AethanFoot's work and lemonaid code to push data asychronously
